src/node/index.ts

```diff
@@ -104,6 +111,40 @@ class Node {
           downloadStream,
         })
 
+        try {
+          const fileBuffer = await readFile(downloadDest);
+          const hashSum = createHash('sha256');
+          hashSum.update(fileBuffer);
+          const fileHash = hashSum.digest('hex');
+          this.logger.info(`Downloaded file ${fileName} SHA256: ${fileHash}`);
+          await utils.download({
+            downloadUrl: hashDownloadUrl,
+            downloadStream: hashDownloadStream
+          });
+          const hashes = await readFile(hashDownloadDest);
+          const hashesContent = hashes.toString();
+          const lines = hashes.toString().split("\n");
+          const shaRegex = /point.*: .{64}/;
+          if (lines.every((line) =>line.match(shaRegex))) {
+            const fileHashes = hashesContent.split("\n").reduce((prev, line) => {
+              const [filename, hash] = line.split(": ");
+              const platform = filename.split('-')[1];
+              return {
+                ...prev,
+                [platform]: hash
+              }
+            }, Object.create(null));
+            const expectedHash = fileHashes[platform]
+            if (fileHash !== expectedHash) {
+              this.logger.error(`File integrity failed sha256 for downloaded file is ${fileHash} but ${expectedHash} was expected`);
```

Comment:
Logging is not what we have to do here, we should do throw and retry the download. In order to do this, let's move the hash checks to a separate function with an interface something like the following.
```
function getExpectedFileSum({pathToSumsFile: string, fileName: string, version: string, platform: string}) -> expectedFileSum: string
function verifyFileSum({filePath, expectedSum}) -> matched: boolean 
```
Both of the functions should throw in case of an error, and we should catch the errors in a retry loop running in `downloadAndInstall` function. This may imply adding more config props for retry logic.

This way we'll be able to test and re-use these functions when downloading `sdk` and the dashboard itself.

Logging is also needed, of course.